### PR TITLE
feat(dbaas): OpenSearch service type support

### PR DESCRIPTION
**Pull request by villevsv-upcloud**

### CODE REVIEW

File: internal/service/database/database.go [hunk 1]
```diff
@@ -375,8 +351,28 @@ func waitServiceNameToPropagate(ctx context.Context, name string) (err error) {
 	return errors.New("max retries reached while waiting for service name to propagate")
 }
 
-func updateDatabaseVersion(ctx context.Context, d *schema.ResourceData, client *service.Service) diag.Diagnostics {
-	diags := diag.Diagnostics{}
+func updateDatabaseVersion(ctx context.Context, d *schema.ResourceData, client *service.Service) (diags diag.Diagnostics) {
```
Comment (peknur) [inline internal/service/database/database.go:354]: I seems that this is now only referenced by PostgreSQL resource, was intentional ?

File: internal/service/database/postgresql.go [hunk 2]
```diff
@@ -27,6 +30,72 @@ func ResourcePostgreSQL() *schema.Resource {
...
+
+func resourcePostgreSQLUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	diags := resourceDatabaseUpdate(ctx, d, meta)
+	if diags.HasError() {
+		return diags
+	}
+
+	client := meta.(*service.Service)
+
+	switch d.HasChange("powered") {
```
[the remark this answers is not in the document]
Comment (villevsv-upcloud) [reply internal/service/database/postgresql.go]: True! Replaced the top level switch and moved the shorter block to be first.

File: internal/service/database/database.go (hunk 1 above)
Comment (villevsv-upcloud) [reply internal/service/database/database.go:354]: It's intentional, as the upgrade operation is only allowed for PostgreSQL resources for now. I opted to use `ForceNew` if user wants to update the version of and OpenSearch resource.

Comment (peknur) [reply internal/service/database/database.go:354]: There is still `version` property in MySQL resource, should we still handle that also ?

Comment (villevsv-upcloud) [reply internal/service/database/database.go:354]: Added `ForceNew: true` to MySQL as well. Currently there is only one whitelisted version (8), so the update issue has not arisen (yet).

**Comment by thevilledev** [comment]
Requires #379 to be merged in first

**Review by peknur — APPROVED** [review]
Looks good :+1:

**Pull request merged**